### PR TITLE
Spun attachments into seperate go file

### PR DESCRIPTION
**Pull request by couling**
Attachments are set to be applied to more things within gitea.  They have already been applied to releases.  

In this PR I've spun them into separate files instead of being burred in `issue.go`.  This will make it easier for future maintainers to work with.

### CODE REVIEW

File: routers/repo/attachment.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
```
Comment (lunny) [inline routers/repo/attachment.go]: 2017 The Gitea

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
Otherwise LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
Otherwise LGTM

**Comment by bkcsoft** [comment]
@lunny I updated the copyright year, just wait for the build to pass

**Pull request merged**